app/models/lookup_key.rb

```diff
@@ -30,7 +30,7 @@ def audit_class
                                 :reject_if => lambda { |a| a[:value].blank? && (a[:use_puppet_default].nil? || a[:use_puppet_default] == "0")},
                                 :allow_destroy => true
 
-  before_validation :validate_and_cast_default_value
+  before_validation :validate_and_cast_default_value, :unless => Proc.new{|p| p[:use_puppet_default]}
```

Comment:
Nitpick, we don't usually use `[]` to access attributes, so `p.use_puppet_default?`